code/modules/mob/living/carbon/human/life.dm

```diff
@@ -276,91 +266,81 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
-					breath_moles = (ONE_ATMOSPHERE*BREATH_VOLUME/R_IDEAL_GAS_EQUATION*environment.temperature)
-				else*/
-					// Not enough air around, take a percentage of what's there to model this properly
-				breath_moles = environment.total_moles()*BREATH_PERCENTAGE
+				if(environment)
+					breath_moles = environment.total_moles()*BREATH_PERCENTAGE
 
 				breath = loc.remove_air(breath_moles)
 
-				if(!is_lung_ruptured())
-					if(!breath || breath.total_moles() < BREATH_MOLES / 5 || breath.total_moles() > BREATH_MOLES * 5)
-						if(prob(5))
-							rupture_lung()
+				if(!is_lung_ruptured())																					// THIS FUCKING EXCERPT, THIS LITTLE FUCKING EXCERPT, SNOWFLAKED
+					if(!breath || breath.total_moles() < BREATH_MOLES / 5 || breath.total_moles() > BREATH_MOLES * 5)	// RIGHT IN THE CENTER OF THE FUCKING BREATHE PROC
+						if(prob(5))																						// IT IS THE ONLY FUCKING REASONS HUMAN OVERRIDE breathe()
+							rupture_lung()																				// GOD FUCKING DAMNIT
 
-		else //Still give containing object the chance to interact
+		else //Breathe from loc as obj again
 			if(istype(loc, /obj/))
-				var/obj/location_as_object = loc
-				location_as_object.handle_internal_lifeform(src, 0)
-
-	handle_breath(breath)
-
-	if(species.name=="Plasmaman") //this is stupid as fuck
-		// Check if we're wearing our biosuit and mask.
-		if (!istype(wear_suit,/obj/item/clothing/suit/space/eva/plasmaman) || !istype(head,/obj/item/clothing/head/helmet/space/eva/plasmaman))
-			//testing("Plasmaman [src] leakin'.  coverflags=[cover_flags]")
-			// OH FUCK HE LEAKIN'.
-			// This was OP.
-			//environment.adjust(tx = environment.total_moles()*BREATH_PERCENTAGE) // About one breath's worth. (I know we aren't breathing it out, but this should be about the right amount)
-			src << "<span class='warning'>Your body reacts with the atmosphere and bursts into flame!</span>"
-			adjust_fire_stacks(0.5)
-			IgniteMob()
+				var/obj/loc_as_obj = loc
+				loc_as_obj.handle_internal_lifeform(src,0)
+
+	check_breath(breath)
 
 	if(breath)
 		loc.assume_air(breath)
-
-		//spread some viruses while we are at it
-		if (virus2.len > 0)
-			if (prob(10) && get_infection_chance(src))
-//					log_debug("[src] : Exhaling some viruses")
+		//spread virus2
+		if(virus2.len > 0)
+			if(prob(10) && get_infection_chance(src))
 				for(var/mob/living/carbon/M in view(1,src))
 					src.spread_disease_to(M)
 
+
+
 // USED IN DEATHWHISPERS
 /mob/living/carbon/human/proc/isInCrit()
 	// Health is in deep shit and we're not already dead
 	return health <= 0 && stat != 2
 
-/mob/living/carbon/human/proc/handle_breath(var/datum/gas_mixture/breath)
+/mob/living/carbon/human/check_breath(var/datum/gas_mixture/breath)
 	if(status_flags & GODMODE)
 		return 0
 
 	if(!breath || (breath.total_moles() == 0) || suiciding)
+		if(reagents.has_reagent("epinephrine"))

```

Comment:
I specifically removed this as it made epinephrine way too good and made the whole "adjust losebreath" redundant on the chemical itself. The only reason it's still on TG is because a blatant "replace" usage of "inaprovaline" and "epinephrine".